sway-lib-std/src/vm/evm/ecr.sw

```diff
@@ -0,0 +1,51 @@
+library ecr;
+
+use ::address::Address;
+use ::b512::B512;
+use ::context::registers::error;
+use ::hash::{HashMethod, hash_pair};
+use ::result::*;
+
+pub enum EcRecoverError {
+    UnrecoverablePublicKey: (),
+}
+
+/// Recover the public key derived from the private key used to sign a message.
+/// Returns a `Result` to let the caller choose an error handling strategy.
+pub fn ec_recover(signature: B512, msg_hash: b256) -> Result<B512, EcRecoverError> {
```

Comment:
Wait why don't you use the `ec_recover` function that already exists for this?

Reply:
Yes, good catch.

Reply:
Resolved by [39e4fba](https://github.com/FuelLabs/sway/pull/1374/commits/39e4fbaaa3481b8acbd05db4744cbb2d4b5f2073)